src/Cli/dotnet/commands/dotnet-workload/install/WorkloadInstallCommand.cs

```diff
@@ -236,6 +267,8 @@ private void WriteSDKInstallRecordsForVSWorkloads()
                         installer.SaveInstallStateManifestVersions(sdkFeatureBand, GetInstallStateContents(manifestsToUpdate));
                     }
 
+                    installer.AdjustWorkloadSetInInstallState(sdkFeatureBand, string.IsNullOrWhiteSpace(_workloadSetVersion) ? null : workloadSetVersion);
```

Comment:
It seems a bit confusing that this uses two different values: `_workloadSetVersion` and `workloadSetVersion`.  Is there any case where `_workloadSetVersion` would be set, but `workloadVersion` would have a different value?  I think not, so I think this can use `_workloadSetVersion` in both cases.  If not, then this should definitely have a comment explaining the difference.

Probably it would be good to have a comment anyway explaining why this uses `_workloadSetVersion` and not `workloadSetVersion`.

Reply:
The only way I think _workloadSetVersion could be set, and it would be different from workloadSetVersion is if the user requests version x, and we reach out to NuGet and ask for x, and it returns version y instead. I don't think that's possible today—I think if NuGet can't find the right version, it would error rather than return the wrong version—so I think you're right that I can switch to just using _workloadSetVersion and delete workloadSetVersion. I should be able to do something similar in WorkloadUpdateCommand.

Reply:
I don't think you want to delete `workloadSetVersion` entirely, just don't use it on this line.  If there's no pinned workload set version and no version was specified on the command line, but the install state is opted into workload sets, then `_workloadSetVersion` would be null, but `workloadSetVersion` would be the latest workload set version that was found, and should be installed / used.

Reply:
Right. We'd still install that version, but we don't actually need to keep track of it in Workload(Update/Install)Command.